thamos/config.py

```diff
@@ -412,7 +412,7 @@ def set_runtime_environment(
             raise
 
         try:
-            existing = self.get_runtime_environment(runtime_environment["name"])
```

Comment:
I think the root of this issue is that `get_runtime_environment` raises the same exception in the case that `runtime_environments` doesn't exist in config, or if the `runtime_environment["name"]` doesn't exist in `runtime_environments`.